app/src/main/java/com/kdani/dagger2_11/MainActivitySubcomponent.java

```diff
@@ -0,0 +1,17 @@
+package com.kdani.dagger2_11;
+
+import com.kdani.dagger2_11.di.scope.PerActivity;
+
+import dagger.Subcomponent;
+import dagger.android.AndroidInjector;
+
+/**
+ * Created by rawsond on 6/11/17.
+ */
+@PerActivity
+@Subcomponent(modules = BaseActivityModule.class)
+public interface MainActivitySubcomponent extends AndroidInjector<MainActivity> {
```

Comment:
So this means I need to manually create subcomponent for every activity??

Reply:
If you want to use a BaseActivity like that I think you will - it seems that the `AndroidInjector` components are not set up very well for that situation. If I think of a better way to get around the problem I will let you know

Reply:
I see. Thanks a lot for specially taking your time for this. I really appreciate your effort :)
For now i'll keep this PR open to see how can we make this scenario more better.
Again thanks a lot for this 👍